src/controller.rs

```diff
@@ -172,13 +179,17 @@ fn clone_resource(
 fn apply_mappings(
     source: &DynamicObject,
     target_ref: &GVKN,
-    target_namespace: &str,
+    target_namespace: Option<&str>,
     ar: &ApiResource,
     sinker: &ResourceSync,
 ) -> Result<DynamicObject> {
-    let mut template = DynamicObject::new(&target_ref.name, ar)
-        .within(target_namespace)
-        .data(json!({}));
+    let mut template = if let Some(ns) = target_namespace {
+        DynamicObject::new(&target_ref.name, ar)
+            .within(ns)
+            .data(json!({}))
+    } else {
+        DynamicObject::new(&target_ref.name, ar).data(json!({}))
+    };
```

Comment:
```suggestion
let mut template = DynamicObject::new(&target_ref.name, ar).data(json!({}));
template.metadata.namespace = target_namespace;
```

Reply:
nice, thanks

Reply:
i did it in a few places so change it in those places too.